### PR TITLE
Added additional header capability to algod client and it's methods

### PR DESCRIPTION
**Pull request by fxgamundi**
Adds support for multiple different headers on the algod client and in all of the internal methods.
Backwards compatible for existing code, both normal Algod string tokens and previous modification that allowed dict tokens.

### CODE REVIEW

**Comment by pzbitskiy** [comment]
@fxgamundi please rebase/resolve conflicts.

**Comment by pzbitskiy** [comment]
@fxgamundi it is hard to review because the diff includes some commits from master. Suggest rebasing on top master and force pushing to have only your 5 commits in the diff.

**Comment by fxgamundi** [comment]
Sorry for that. Now everything should be clean!

File: src/client/client.js [hunk 1]
```diff
@@ -14,17 +14,21 @@ function removeEmpty(obj) {
...
 
-function HTTPClient(requestHeaders, baseServer, port) {
+function HTTPClient(token, baseServer, port, headers={}) {
     // Do not need colon if port is empty
     if (port !== '') {
         baseServer += ":" + port.toString();
     }
     this.address = baseServer;
+    this.token = token;
+    this.clientHeaders = headers;
```
Comment (rotemh) [inline src/client/client.js]: Can we call it `DefaultHeaders`? Client seems a bit off to me in this context.

**Pull request merged**